### ISSUE TITLE
Fix LoRA training and long renders on 48 GB Macs

### ISSUE THREAD

**Issue by anubissbe** [mrbizarro/phosphene#30]
## Problem

On 48 GB Apple Silicon Macs, LoRA training and long Q4 video generation can enter severe memory pressure and swap thrash.

Observed symptoms:
- Character LoRA training showed multi-hour ETAs for "Quick" runs.
- A 47-49 image dataset could create hundreds/thousands of full training steps depending on preset and stale form values.
- Long I2V generation at 1024x576, 481 frames, and large LoRA stacks stayed slow even after competing work completed.
- Auto-captioning with Gemma could run while a GPU render was already active, increasing unified-memory pressure.

## Root Cause

The panel grouped 48 GB and 64 GB systems into the same generation/training assumptions. That is workable for many inference paths, but too optimistic for LoRA backprop and long native-frame renders with large LoRA stacks.

The server also trusted stale advanced Train form values, so old browser state or replayed params could bypass safer UI defaults.

## Proposed Fix

Add hardware-specific profiles for 48 GB machines:
- Compact LoRA training presets with server-side clamps for rank, steps, resolution, and target modules.
- Honest Train tab bootstrap data from the server instead of hardcoded JS estimates.
- Generation profile that auto-enables Long Clip Boost for long Q4 T2V/I2V jobs and clamps oversized Standard dimensions.
- Block Gemma auto-captioning while any GPU job is active.

## Validation

- `python3 -m py_compile mlx_ltx_panel.py lora_lab/train_character.py`
- `git diff --check`
- Dry-run job construction for the observed bad cases.
- Live run confirmed training progressed from step 10/490 to 60/490 with ETA dropping to ~34.5 minutes.

**Comment by mrbizarro** [comment]
Thanks for the thorough writeup — this matches real behavior on 48 GB unified memory, and it's a real gap: the panel did group 48 GB and 64 GB into the same assumptions. Your PR #31 is a well-scoped fix and I'm reviewing it now — the server-side clamps, honest Train-tab bootstrap, and especially the "no Gemma auto-captioning while a live GPU job is running" guard are things I want regardless of profile.

I'll follow up with review notes on the PR. Keeping this open until the fix lands.

**Comment by mrbizarro** [comment]
The fix (@anubissbe's PR #31) is adopted onto the release branch now — 48 GB training + long-render profiles with server-side clamps. It'll land in the next release. Closing #31 as adopted; leaving this open until the release ships.

**Comment by mrbizarro** [comment]
@anubissbe — your fix is on the dev branch and ships with the next release. Since this issue is the thing people will find when they search "48 GB", here's exactly what the code does today, including a wrinkle in your diagnosis that's worth writing down.

**There are two independent tier systems, and 48 GB lands on opposite sides of them.**

The capability tier (what the UI exposes) buckets by total RAM: under 48 → `base`, 48–79 → `standard`, 80–119 → `high`, 120+ → `pro`. A 48 GB Mac is `standard` ("Comfortable"), which sets `allows_q8: true`, so the panel runs at the **full `q8` capability surface** — nothing is hidden.

The *job-shaping* profiles are a separate branch on `< 64 GB`, and that's the one that bites. So a 48 GB machine is simultaneously top-half for what it's allowed to open and "compact" for how its jobs get built. That mismatch is the real content of your report — the panel didn't group 48 and 64 together everywhere, it grouped them in the UI and split them nowhere else. Now it splits them in job construction too.

**What a 48 GB Mac can do today**

- **Every mode is available**: High / Q8 quality, keyframe (FFLF), Extend, characters, lip-dub. No `display:none` gates apply at `q8`.
- **T2V / I2V**: no tier clamp on dimensions, but the 48 GB generation profile clamps non-High renders to a 1024 max side (32-aligned). **High is deliberately left alone** — it uses a different sampler and doesn't support the 12→24 fps path, so clamping it would change output, not just cost.
- **Long Clip Boost auto-enables** for native-fps T2V/I2V past 241 frames (~10 s) instead of letting you walk into the slow path.
- **Keyframe / Extend** are capped at 768 px on this tier.
- **LoRA stacks over 6** produce a warning note, not a block.
- **Hailuo H3 is off**: it needs 60 GB and 48 doesn't clear the floor.

**Training on 48 GB** runs the compact profile: rank ≤ 8, resolution ≤ 448, steps ≤ 500, and two target modules (`to_q`, `to_v`) instead of four. Presets are Quick 2 epochs / rank 4 / 384 px, Medium 5 / rank 8 / 384, High 10 / rank 8 / 448, with documented peaks of ~38 / 40 / 44 GB — those peaks are the actual reason for the ceilings. For contrast, ≥64 GB "High" is 100 epochs / rank 32 / 512 px / 20 000 steps, which is what your multi-hour ETAs were quoting at you. **Voice/audio training isn't tier-gated at all** and runs the same on any machine.

The clamps are enforced **server-side in job construction**, exactly as you argued — a stale browser tab replaying old rank-32/10 000-step values gets clamped and told so in the job notes, rather than accepted and swap-thrashed. The Train tab now bootstraps its presets and ceilings from the server instead of hardcoded JS, and out-of-range controls are disabled rather than silently corrected. And Gemma auto-captioning is refused with a 409 while any GPU job is active — that guard was the one I wanted regardless of RAM.

Escape hatches, so nobody hunts for them: `LTX_TIER_OVERRIDE` (`base|standard|high|pro`) and `LTX_FORCE_CAP_TIER` (`q4|q8`) exist for testing the UI surfaces, but **neither lifts the 64 GB job-shaping boundary** — those profiles branch on measured RAM and don't read the tier. That's intentional for now, but it does mean a 48 GB user who knows what they're doing can't opt into a bigger training run, and I don't love that.

Lower-memory support is a real axis I care about — the honest position is that the presets and the memory floors were tuned on 64 GB and everything below that is inference-first with training as the tight case. I'm not going to promise a specific expansion of the 48 GB envelope, but if you have runs where the compact ceilings are leaving headroom on the table (peak RAM well under those 38/40/44 numbers), that's exactly the data that would move it — please post it here.

Keeping this open until the release ships.

**Comment by mrbizarro** [comment]
Closing this — @anubissbe's PR #31 shipped publicly some time ago and I never came back to say so. Correcting that now, with the check done against the tagged tree rather than from memory.

Everything in the proposed fix is present in public **v3.3.0** and every release since (current is v3.7.0):

- `TRAIN_PROFILE` — compact LoRA training presets for 48 GB-class Macs, with the server-side clamps on rank / steps / resolution / target modules, so stale advanced form values can no longer bypass them.
- `GENERATION_PROFILE` — job-shaping profile that clamps oversized non-High renders and records what it changed in `generation_clamp_notes`, so the panel tells you it clamped rather than silently doing it.
- `auto_temporal_after_frames` — Long Clip Boost auto-enables past ~10 s of native-fps T2V/I2V instead of letting you walk into the slow path.
- The auto-caption handler refuses to start while **any** GPU job is in flight, not just training — which was the specific hole that let Gemma run beside a long render on a 48 GB machine.

One thing worth leaving on the record, because it is the part of the diagnosis that was slightly off and it is what people will find when they search "48 GB": there are two independent tier systems and 48 GB lands on opposite sides of them. The *capability* tier buckets 48–79 GB as `standard`, which sets `allows_q8: true` — so nothing is hidden in the UI. The *job-shaping* profile branches on < 64 GB. So a 48 GB Mac is top-half for what it may open and "compact" for how its jobs get built. The panel didn't group 48 and 64 together everywhere; it grouped them in the UI and split them nowhere else. Now it splits them in job construction too.

Thanks @anubissbe — the server-side clamps and the caption/GPU guard were worth having regardless of the profile work.
